File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/listener/CallUiListener.java
package com.stringee.apptoappcallsample.stringee.listener;

import com.stringee.apptoappcallsample.stringee.common.CallStatus;
import com.stringee.common.StringeeAudioManager.AudioDevice;
import com.stringee.video.StringeeVideoTrack;

import java.util.Set;

/** Internal callbacks used to render an owned {@code CallSession} in the call activity. */
public interface CallUiListener {
    void onCallStatus(CallStatus status);

    void onError(String message);

    void onLocalVideoAvailable();

    void onRemoteVideoAvailable();

    void onAudioDeviceChanged(AudioDevice selected, Set<AudioDevice> available);

    void onMicChanged(boolean enabled);

    void onVideoChanged(boolean enabled);

    void onSharingChanged(boolean sharing);

    void onTimer(String duration);

    void onScreenTrackAdded(StringeeVideoTrack track);

    void onScreenTrackRemoved(StringeeVideoTrack track);
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/listener/CallUiListener.kt
package com.stringee.kotlin_onetoonecallsample.stringee.listener

import com.stringee.common.StringeeAudioManager.AudioDevice
import com.stringee.kotlin_onetoonecallsample.stringee.common.CallStatus
import com.stringee.kotlin_onetoonecallsample.stringee.manager.CallSession
import com.stringee.video.StringeeVideoTrack

/** Internal callbacks used to render an owned [CallSession] in the call activity. */
interface CallUiListener {
    fun onCallStatus(status: CallStatus)

    fun onError(message: String?)

    fun onLocalVideoAvailable()

    fun onRemoteVideoAvailable()

    fun onAudioDeviceChanged(selected: AudioDevice?, available: MutableSet<AudioDevice?>?)

    fun onMicChanged(enabled: Boolean)

    fun onVideoChanged(enabled: Boolean)

    fun onSharingChanged(sharing: Boolean)

    fun onTimer(duration: String?)

    fun onScreenTrackAdded(track: StringeeVideoTrack?)

    fun onScreenTrackRemoved(track: StringeeVideoTrack)
}
